src/components/molecules/AnnouncementMessage/AnnouncementMessage.scss

```diff
@@ -15,7 +15,7 @@
     left: 50%;
   }
 
-  padding: 15px;
+  padding: 15px 30px 15px 20px;
```

Comment:
It would be good to use some sass `$` variables here, and pass that to the `.close-button` `right` attribute so that we can guarantee they will stay 'in sync'.

Looking at the screenshot in https://github.com/sparkletown/sparkle/pull/1184#issue-574472150 it feels almost 'too spacey'. The overly rounded corners we use throughout our styling at the moment make this somewhat challenging to make look good IMO, but perhaps we could reduce the spacing somewhat as well?

Playing around locally:

- top: `10px` (shared variable with `.close-button`)
- bottom: `10px` (can probably use the same variable, but doesn't need to be applied to `.close-button`'s bottom padding
- (i forget if there were other bits and pieces here as I switched to playing with the below)

If it was me doing things, I would probably even go as far as changing the `border-radius` from `50px` down to like `20px` or so. (you would want to check with @sparkletown/product and/or @sparkletown/design for that), which would then allow the styles to be refined even better:

- `.close-button` could have top `10px`, right `7.5px`
- `.announcement-container` would have `padding: 10px 0px`
- we'd add a new `div` that wraps the text (but not the close icon?) and add styles to set margin left/right `30px`
  - (honestly not sure why 30px.. with the close icon positioned `10px` from the right, I would have expected a right margin of `15px` would give sufficient space, but if you look at the orange areas in the screenshot below (margins) when set to `30px` it seems to only just be enough space.)
    - ![image](https://user-images.githubusercontent.com/753891/108448901-ad2db400-72b6-11eb-8881-d57d198ff295.png)
  - (It might have to do with the width of the close icon, since that seems to be `20px`, so adding the `7.5px` right position, that would be `27.5px`; so `30px` ensures we have at least `2.5px` of space between the edge of the text, and the close icon)
    - ![image](https://user-images.githubusercontent.com/753891/108449269-4e1c6f00-72b7-11eb-9e73-3ef1fd272824.png)

The logic in `src/utils/getLinkFromText.tsx` that wraps every word in a `span` makes it hard to play around with this in the Chrome DevTools. That could be changed to use `React.Fragment` (on line 22), which would then unclutter things and make it easier to play with.

Further Reading:

- https://sass-lang.com/documentation/variables
- https://sass-lang.com/documentation/operators/numeric
- https://sass-lang.com/documentation/syntax/special-functions#calc-clamp-element-progid-and-expression
  - I think the numeric operators above should be sufficient, but if not, this reference may be useful
  
And for my reference, since I always forget the order/which it refers to:

- https://www.w3schools.com/css/css_padding.asp
  - 4: top, right, bottom, left
  - 3: top, right/left, bottom
  - 2: top/bottom, right/left
  - 1: top/bottom/right/left

Reply:
(which was way more detail/thought than I intended to put into that.. take the specifics with a grain of salt, but the main intent here was to reduce the 'overly padded' feeling a bit, and to use variables to share various widths/etc used to calculate the numbers we use (so that it's clear how they were calculated + will auto update themselves if we change any of the 'parts of the calculation' in future)

Reply:
Should I also change the class names to follow the BEM approach?

Reply:
10px on the top and bottom looks really awkward when the banner is only 1 line, so I'm leaving it at 15px. 

10px:
<img width="723" alt="Screen Shot 2021-02-19 at 10 05 00 AM" src="https://user-images.githubusercontent.com/51083763/108489859-fa4f5d00-7299-11eb-842e-10c88dd3e108.png">

15px:
<img width="690" alt="Screen Shot 2021-02-19 at 10 05 46 AM" src="https://user-images.githubusercontent.com/51083763/108489944-13580e00-729a-11eb-82e9-fa9eaee761c7.png">

Reply:
I think 15px looks good even when it's two lines:
<img width="748" alt="Screen Shot 2021-02-19 at 10 06 32 AM" src="https://user-images.githubusercontent.com/51083763/108490056-31be0980-729a-11eb-8efa-6bbb847d91a2.png">

Reply:
I’m ok with 15px